More fixes

nirbheek opened the pull request:
commit a5e198de784dd07f358dac8e40a9dcc9e93b1029:

Suppress warnings with gcc and clang

commit e045d9f1f9e108402c3874fd6c6050050527795e:

Add a new option for a more minimal build `-Dsolo=true`

nirbheek committed:
Suppress warnings with gcc and clang

nirbheek committed:
Add a new option for a more minimal build

sarum9in commented on meson.build:
```diff
@@ -6,9 +6,15 @@ link_args = []
-  vflag = '-Wl,--version-script,@0@/zlib.map'.format(meson.current_source_dir())
-  link_args += [vflag]
+else
+  # Don't spam consumers of this wrap with these warnings
```
Maybe we just disable warnings as a target option via `override_options`?

nirbheek replied:
Not sure I understand what you mean. Do you mean `warning_level:` on the `library()` target? That's not the only way these warnings can be enabled, so this method would be more fool-proof.

sarum9in replied:
Yes, I meant that. Are you talking about users overriding options manually? Given that both options will be present is there any guarantee that option set here will take preference?

nirbheek replied:
Oh, hmm, I thought I added the compiler args to the target but I didn't. Arguments added to a target with `compile_args:` override everything else, so let me fix that.

nirbheek replied:
Should be fixed now!

nirbheek committed:
Disable warnings by adding them to the target c_args

This way we can be sure that they won't crop up no matter how extra
warning args are enabled by the user.
